pdns/pdnsutil.cc

```diff
@@ -564,6 +564,17 @@ int checkZone(DNSSECKeeper &dk, UeberBackend &B, const DNSName& zone, const vect
       }
     }
 
+    if(rr.qtype.getCode() == QType::A || rr.qtype.getCode() == QType::AAAA)
+    {
+      Regex hostnameRegex=Regex("^(([A-Za-z0-9]([A-Za-z0-9-]*[A-Za-z0-9])?)\\.)+$");
+      if (!hostnameRegex.match(rr.qname.toString()))
+      {
+        cout<<"[Error] A or AAAA record found at '"<<rr.qname.toString()<<"'. This name is not a valid hostname."<<endl;
```

Comment:
@Habbie is it only https://tools.ietf.org/html/rfc952 + https://tools.ietf.org/html/rfc1123#section-2 that you had in mind (defines what is a valid hostname) or also something to support the inference that a name with address records should be a hostname?  Also, is this for a code comment or did you want the RFC references in the output?